.github/workflows/testing_and_building_repo.yml

```diff
@@ -0,0 +1,59 @@
+name: testing_and_building_repo
+on: [push, pull_request]
+jobs:
+  build: 
+    strategy:
+      matrix:
+        os: ['ubuntu-18.04']
+        
+    runs-on: ${{matrix.os}} 
+    name: Distribution Perl on ${{ matrix.os }} 
+    steps:
+    - uses: actions/checkout@v2 
+    
+   # Caching cpanm external modules
+    - name: Cache cpanm external modules
+      id: cpanmCache
+      uses: actions/cache@v2
+      with:
+        path: ~/perl5ext  
+        key: ${{ matrix.os}}-build-cpanm-external
+ 
+    - name: install cpanm 
+      run: |
+          wget -qO - https://cpanmin.us | /usr/bin/perl - --sudo App::cpanminus
+
+    # run the perl install script
+    - name: install NPG Perl dependencies, and their CPAN dependencies
+      run: |
+          cpanm --local-lib=~/perl5ext local::lib && eval $(perl -I ~/perl5ext/lib/perl5/ -Mlocal::lib)
+          ${GITHUB_WORKSPACE}/scripts/install_npg_perl_dependencies.sh $WTSI_NPG_GITHUB_URL $WTSI_NPG_BUILD_BRANCH
+      env:
+        WTSI_NPG_GITHUB_URL: https://github.com/wtsi-npg 
+        WTSI_NPG_BUILD_BRANCH: ${GITHUB_HEAD_REF}
+      
+    - name: install cpanm dependencies
+      run: |
+          eval $(perl -I ~/perl5ext/lib/perl5/ -Mlocal::lib=~/perl5ext)
+          cpanm --installdeps --notest .
+
+    # running ./Build dist and exporting files
+    - name: run ./Build dist  
+      run: |
+          eval $(perl -I ~/perl5ext/lib/perl5/ -Mlocal::lib=~/perl5ext)
+          export TEST_AUTHOR=1
+          ./Build dist
+          export DIST_FILE=$(ls ml_warehouse-*.tar.gz)
```

Comment:
You don't need a sub-shell and `ls` for the glob:

```
$ touch foo_1234.txt
$ x=foo_*.txt
$ echo "$x"
foo_1234.txt

Reply:
So no `ls`, just :
`export DIST_FILE=$(ml_warehouse-*.tar.gz)` ?

Reply:
Not even that, see the example above. The subshell isn't needed.
`export x=foo_*.txt`

(There is a case for using a subshell if you use e.g. `shopt -s nullglob` to trigger a failure if the glob doesn't match anything and also want to confine the effect of the `shopt` to just the shell where the glob is done)